implementations/rust/message/src/lib.rs

```diff
@@ -1,517 +1,593 @@
+#![allow(unused)]
+
 // Definition and implementation of an Ockam message and message components.
 // Each message component, and the message overall, implements the "Codec" trait
 // allowing it to be encoded/decoded for transmission over a transport.
 
-use std::{
-    convert::TryFrom,
-    fmt::Formatter,
-    io::ErrorKind,
-    net::{IpAddr, Ipv4Addr, SocketAddr},
-    str::FromStr,
-};
+pub mod message {
```

Comment:
Now that the file is in `message/src/lib/rs` we should omit the `pub mod message` scope wrapping the message code. This is the "path shortening" concern @mikelodder7 mentioned. 

From a user standpoint, including this mod wrapper will require typing `use ockam_message::message::Message` rather than `use ockam_message::Message`. 

I'd suggest we remove the mod wrapper, but am open to alternate feedback.

Reply:
Issue opened